src/layout/page-layout.scss

```diff
@@ -0,0 +1,384 @@
+    }
+  }
+
+  &.PageLayout--outerSpacing-condensed {
+    --Layout-outer-spacing-x: #{$spacer-3}; // 16px
+    --Layout-outer-spacing-y: #{$spacer-3}; // 16px
+  }
+
+  // inner spacing
+
+  &.PageLayout--innerSpacing-normal {
+    --Layout-inner-spacing-min: #{$spacer-3}; // 16px
+    --Layout-inner-spacing-max: #{$spacer-3}; // 16px
+
+    @include breakpoint(lg) {
+      --Layout-inner-spacing-max: #{$spacer-4}; // 24px
+    }
+  }
+
+  &.PageLayout--innerSpacing-condensed {
+    --Layout-inner-spacing-min: #{$spacer-3}; // 16px
+    --Layout-inner-spacing-max: #{$spacer-3}; // 16px
+  }
+
+  // column gap
+
+  &.PageLayout--columnGap-normal {
+    --Layout-column-gap: #{$spacer-3}; // 16px
+
+    @include breakpoint(lg) {
+      --Layout-column-gap: #{$spacer-4}; // 24px
+    }
+  }
+
+  &.PageLayout--columnGap-condensed {
+    --Layout-column-gap: #{$spacer-3}; // 16px
+  }
+
+  &.PageLayout--columnGap-none {
+    // the `px` unit is mandatory for `calc()` execution. See https://stackoverflow.com/a/32518348
+    // stylelint-disable-next-line length-zero-no-unit
+    --Layout-column-gap: 0px;
+  }
+
+  // row gap
+
+  &.PageLayout--rowGap-normal {
+    --Layout-row-gap: #{$spacer-3}; // 16px
+
+    @include breakpoint(lg) {
+      --Layout-row-gap: #{$spacer-4}; // 24px
+    }
+  }
+
+  &.PageLayout--rowGap-none {
+    // the `px` unit is mandatory for `calc()` execution. See https://stackoverflow.com/a/32518348
+    // stylelint-disable length-zero-no-unit
+    --Layout-row-gap: 0px;
+  }
+
+  &.PageLayout--rowGap-condensed {
+    --Layout-row-gap: #{$spacer-3}; // 16px
+  }
+
+  // regions
+
+  .PageLayout-header,
+  .PageLayout-content,
+  .PageLayout-pane,
+  .PageLayout-footer {
+    // stylelint-disable-next-line primer/spacing
+    padding: var(--Layout-inner-spacing-min);
+  }
+
+  .PageLayout-content {
+    // stylelint-disable primer/spacing
+    padding-right: var(--Layout-inner-spacing-max);
+    padding-left: var(--Layout-inner-spacing-max);
+    // stylelint-enable primer/spacing
+    grid-area: content;
+  }
+
+  .PageLayout-pane {
+    grid-area: pane;
+  }
+
+  // multi-column desktop-friendly layout
+
+  @include breakpoint($Layout-responsive-variant-max-breakpoint) {
+
+    // Set a `content` region width, to work with loading states when
+    // `pane` is not yet loaded. See https://github.com/primer/css/pull/1818
+    $Layout-content-full-width: minmax(0, calc(100% - var(--Layout-pane-width) - var(--Layout-column-gap)));
+
+    &.PageLayout--panePos-start {
+      --Layout-template-columns: var(--Layout-pane-width) #{$Layout-content-full-width};
+      --Layout-template-areas: 'pane content';
+    }
+
+    &.PageLayout--panePos-end {
+      --Layout-template-columns: #{$Layout-content-full-width} var(--Layout-pane-width);
+      --Layout-template-areas: 'content pane';
+    }
+
+    // header divider
+
+    &.PageLayout--hasHeaderDivider {
+      .PageLayout-header {
```

Comment:
Instead of adding these styles by using a descendant selector, why not just create a modifier class directly on `.PageLayout-header`?

For example:
```css
.PageLayout-header--hasDivider {
        // stylelint-disable-next-line primer/borders
        border-bottom: $border-width solid $Layout-divider-color;
        // stylelint-disable-next-line primer/spacing
        padding-bottom: max(var(--Layout-row-gap), var(--Layout-inner-spacing-min));
}
```

Reply:
@mperrotti I think I did that because for the pane divider it needed to be set at the root level, as `.PageLayout--hasPaneDivider` (since it influences other parts the layout such as the content centered width position. Do you think it'd be weird to have `--hasPaneDivider` in the root but `--hasDivider` in the `header`/`footer` regions?

Reply:
Moved `--hasDivider` to `PageLayout-header`. ✅